Reorganize content in log management overview doc, fix/clean wording. (#2953)
quickwit-oss/quickwit: docs/log-management/index.md

---
title: Log management with Quickwit
sidebar_position: 1
---

import DocCardList from '@theme/DocCardList';

> Quickwit is built from the ground up to [index efficiently unstructured data](../guides/schemaless.md), and search directly your data sitting on cloud storage.

Reviewer: ```suggestion
Quickwit is built from the ground up to [efficiently index unstructured data](../guides/schemaless.md), and search it effortlessly on cloud storage.
```

Maybe better way of wording so it reads nicer.

> Moreover, Quickwit is OpenTelemetry native and provides a REST API ready to ingest any JSON formatted logs.

Reviewer: ```suggestion
Moreover, Quickwit supports OpenTelemetry out of the box and provides a REST API ready to ingest any JSON formatted logs.
```

**This makes Quickwit a perfect fit for logs!**.

![Quickwit Log Management Overview](../assets/images/logs-management-overview.svg)

## Sending logs to Quickwit

<DocCardList />

## Supported agents

### OpenTelemetry agent

Before using an [OpenTelemety collector](https://opentelemetry.io/docs/collector/), check that [Quickwit OpenTelemetry service](#opentelemetry-service) is enabled.

> Once started, Quickwit is then ready to receive OpenTelemetry gRPC requests.

Reviewer: ```suggestion
Once started, Quickwit is then ready to receive and ingest OpenTelemetry gRPC requests.
```

> Here is configuration example of an OpenTelemetry agent that sends logs into Quickwit:

Reviewer: ```suggestion
Here is a configuration example of an OpenTelemetry agent that sends logs into Quickwit:
```

config:
  exporters:
    otlp:

>       # Local instance localhost:7281

Reviewer: When first reading this I'm not really sure what this means, is the config setup for k8s but this comment saying you can use localhost if running locally?

Author: I need to be clearer here indeed.

Author: Could be any host, on k8s you will want a url of the form `{quickwit-indexer-service-name}.{namespace}.svc.cluster.local:7281`

          - otlp
```

> Find more configuration details on the [OpenTelemetry documentation](https://opentelemetry.io/docs/collector/configuration/). You can also follow our [tutorial to send Kubernetes logs](deploy-quickwit-otel-with-helm.md) to Quickwit.

Reviewer: ```suggestion
Find more configuration details on the [OpenTelemetry documentation](https://opentelemetry.io/docs/collector/configuration/). You can also check out our [tutorial to send Kubernetes logs](deploy-quickwit-otel-with-helm.md) to Quickwit.
```

## UI Integration

Currently, Quickwit provides a simplistic UI to get basic information from the cluster, indexes and search documents.

> If a simple UI is not sufficient for you and you need additional features, Grafana and Elasticsearch query API support are planned for Q2 2023, stay tuned!

Reviewer: Is the jaeger support not something which may be interesting to put here?

Author: Not for logs. I have a separated PR for distributed tracing here: https://github.com/quickwit-oss/quickwit/pull/2947
But not yet ready for review.

Reviewer: Hmm maybe worth associating the two pages then, however, if I'm reading through everything it may pique my interest to move on to the distributed tracing stuff and vice versa.

Author: Good point indeed.
